src/js/iotjs.js

```diff
@@ -196,56 +197,41 @@
   };
 
   function prepareStackTrace(throwable) {
-    return [
-      CallSite.create('main', ['<anonymous>', 1, 7]),
-      CallSite.create('main', ['<anonymous>', 2, 7]),
-      CallSite.create('main', ['<anonymous>', 3, 7]),
-      CallSite.create('main', ['<anonymous>', 4, 7]),
-    ];
+    return makeStackTraceFromDump(throwable.__frames__ || []);
   }
 
+  var stackPropertiesDescriptor = {
+    stack: {
+      configurable: true,
+      enumerable: false,
+      get: function () {
+        if (this.__stack__ === undefined) {
+          process._flushParserDumpFile();
+          this.__stack__ = makeStackTraceFromDump(this.__frames__);
+        }
+        return this.__stack__;
+      },
+    },
+    __stack__: {
+      configurable: true,
+      writable: true,
+      enumerable: false,
+    },
+  };
+  Object.defineProperties(Error.prototype, stackPropertiesDescriptor);
```

Comment:
If the properties are not attached to the prototype, how can we get all error instance got the property `stack` without modifying native `Error` constructor and sharing the same stack generating code in JS?
The prototype changing of native types has to be prevented in cases of implementing our own API, however `error.stack` is a commonly used property in JS, it might be ok.

Reply:
Good point, a false `enumerable` value is reasonable and you has added it in the latest patch :p